diff --git a/scripts/cocoapods/codegen.rb b/scripts/cocoapods/codegen.rb
index 56d120fcceda051e8bdda6f0af63dcf2dd16d4a6..8be6c547581b44733d7771394622cdf7f35dcdd2 100644
--- a/scripts/cocoapods/codegen.rb
+++ b/scripts/cocoapods/codegen.rb
@@ -11,14 +11,18 @@
 # - dir_manager: a class that implements the `Dir` interface. Defaults to `Dir`, the Dependency can be injected for testing purposes.
 # @throws an error if it could not find the codegen folder.
 def build_codegen!(react_native_path, relative_installation_root, dir_manager: Dir)
-    codegen_repo_path = "#{relative_installation_root}/#{react_native_path}/../react-native-codegen";
-    codegen_npm_path = "#{relative_installation_root}/#{react_native_path}/../@react-native/codegen";
+    react_native_relpath = "#{relative_installation_root}/#{react_native_path}"
+    codegen_repo_path = "#{react_native_relpath}/../react-native-codegen";
+    codegen_npm_path = "#{react_native_relpath}/../@react-native/codegen";
+    codegen_pnpm_path = "#{react_native_relpath}/node_modules/@react-native/codegen";
     codegen_cli_path = ""
 
     if dir_manager.exist?(codegen_repo_path)
       codegen_cli_path = codegen_repo_path
     elsif dir_manager.exist?(codegen_npm_path)
       codegen_cli_path = codegen_npm_path
+    elsif dir_manager.exist?(codegen_pnpm_path)
+      codegen_cli_path = codegen_pnpm_path
     else
       raise "[codegen] Could not find react-native-codegen."
     end
